### DISCUSSION TITLE
Question: Inserting a New Line.

### DISCUSSION THREAD

**Discussion by nrikiji** [facebook/lexical#3423] [Q&A] [answered]
Is it possible to insert a line that preserves the status of the previous line when a new line is inserted with the enter key? If you have a sample code, could you please let me know?

That is not the behavior in Lexical's Playground. Also, the Draft.js demo has that behavior.

**Comment by zignis** [comment]
> Is it possible to insert a line that preserves the status of the previous line when a new line is inserted with the enter key? If you have a sample code, could you please let me know?
> 
> That is not the behavior in Lexical's Playground. Also, the Draft.js demo has that behavior.

You can try line break node or `$createTextNode` with `$createParagraphNode` (kind of hack):
```tsx
import {
  $createTextNode,
  $createParagraphNode,
  $insertNodes,
} from 'lexical';

...

const textNode = $createTextNode('\n');
const paragraphNode = $createParagraphNode();
paragraphNode.append(textNode);

$insertNodes([paragraphNode]);
```

**Comment by acywatson** [comment]
> Is it possible to insert a line that preserves the status of the previous line when a new line is inserted with the enter key? If you have a sample code, could you please let me know?
> 
> That is not the behavior in Lexical's Playground. Also, the Draft.js demo has that behavior.

When you say "preserves the status" what do you mean, exactly?

**Comment by nrikiji** [comment]
@zignis 
Thanks for the reply. Is it like adding a node by handling the enter key around the following?

```ts
editor.registerUpdateListener(({ editorState }) => {
    ・・・
});
・・・
editor.update(() => {
    ・・・
});
```

**Comment by nrikiji** [comment]
@acywatson 
Thanks for the reply. For example, we want to continue the selected font size and color in the next line.

**Comment by zignis** [comment, marked as answer]
> @zignis Thanks for the reply. Is it like adding a node by handling the enter key around the following?
> 
> ```ts
> editor.registerUpdateListener(({ editorState }) => {
>     ・・・
> });
> ・・・
> editor.update(() => {
>     ・・・
> });
> ```

Yes, you may want to take a look at how the `@lexical/code` package does it: https://github.com/facebook/lexical/blob/main/packages/lexical-code/src/CodeHighlighter.ts#L515

**Reply by nrikiji** [reply]
@zignis 
https://lexical.dev/docs/concepts/commands
Thank you. This seems like a good solution.
